### PR TITLE
Fixed code to pass CryptoGetInfoRegression.

### LINKED ISSUE #9194: Start testing CryptoGetInfoRegression

### PR DESCRIPTION
**Pull request by iwsimon**
**Description**:

Fixed code to pass HapiTests in CryptoGetInfoRegression.
Removed unused recordBuilder in SystemFileUpdateFacility.handleTxBody(),
Removed status == PLATFORM_NOT_ACTIVE check in FeesAndRatesProvider.downloadWith().

**Related issue(s)**:

Fixes #9194

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by povolev15 — DISMISSED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java [hunk 1]
```diff
@@ -296,7 +297,7 @@ private void checkTransactionDeprecation(@NonNull final Transaction tx) throws P
             // If they didn't use `signedTransactionBytes` and they didn't use `bodyBytes` then they didn't send a body
             // NOTE: If they sent a `sigMap` without a `bodyBytes`, then the `sigMap` will be ignored, just like
             // `body` and `sigs` are. This isn't really nice but not fatal.
-            throw new PreCheckException(INVALID_TRANSACTION_BODY);
+            throw new PreCheckException(NOT_SUPPORTED);
```
Comment (kimbor) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java:300]: This doesn't seem right. If there is no transaction body then it seems like INVALID_TRANSACTION_BODY is the correct response. NOT_SUPPORTED is supposed to be for when the transaction or query type is deprecated or never implemented.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java [hunk 2]
```diff
@@ -77,13 +76,9 @@ public SystemFileUpdateFacility(
      * @param state the current state (the updated file content needs to be committed to the state)
      * @param txBody the transaction body
      */
-    public void handleTxBody(
-            @NonNull final HederaState state,
-            @NonNull final TransactionBody txBody,
-            @NonNull final SingleTransactionRecordBuilderImpl recordBuilder) {
+    public void handleTxBody(@NonNull final HederaState state, @NonNull final TransactionBody txBody) {
```
Comment (kimbor) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java:79]: It looks like Valentin just added the SingleTransactionRecordBuilderImpl argument in PR #8219 that was merged three weeks ago. I wonder why he added it, as you noticed it is not actually being used.

Comment (vtronkov) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java:79]: As far as I can remember I had to do some refactoring and this field was a miss. Good catch on cleaning that up 🏆

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/fees/FeesAndRatesProvider.java [hunk 3]
```diff
@@ -190,7 +189,7 @@ private FileGetContentsResponse downloadWith(long queryFee, boolean costOnly, Fi
                     .getFileContent(query)
                     .getFileGetContents();
             status = response.getHeader().getNodeTransactionPrecheckCode();
-            if (status == OK || status == PLATFORM_NOT_ACTIVE) {
```
Comment (vtronkov) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/fees/FeesAndRatesProvider.java:193]: Why this change was needed? Looks like @jsync-swirlds added this check(https://github.com/hashgraph/hedera-services/pull/9197/files#diff-9bbb90112981633f33079ee75e18634d4dbab99a634c63baa6a119b17a892117R193) two days ago.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java:300]: This is to follow mono service behavior. INVALID_TRANSACTION_BODY failed on one of the test in this suite.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java (hunk 2 above)
Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java:79]: Yes, it's not used anywhere.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/fees/FeesAndRatesProvider.java (hunk 3 above)
Comment (iwsimon) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/fees/FeesAndRatesProvider.java:193]: @jsync-swirlds has his pr 9212 to remove this as well and already merged. I will merge from develop.

**Review by vtronkov — DISMISSED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java:300]: @iwsimon Did you see logic somewhere that threw `NOT_SUPPORTED` ? 
I see [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/submission/StructuralPrecheck.java#L88) it throws `INVALID_TRANSACTION_BODY`

**Review by Neeharika-Sompalli — COMMENTED** [review]
Have one question. Other changes look fine to me.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java (hunk 1 above)
Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java:300]: @Neeharika-Sompalli this test:
failsForMissingPayment() {
        return defaultHapiSpec("FailsForMissingPayment")
                .given()
                .when()
          .then(getAccountInfo(GENESIS).useEmptyTxnAsAnswerPayment().hasAnswerOnlyPrecheck(NOT_SUPPORTED));
    }
 expects NOT_SUPPORTED.

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java:300]: Actually that test is failing in mono-service also. It should have `INVALID_TRANSACTION_BODY` 😄

Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java]: Set back to INVALID_TRANSACTION_BODY and updated the test with INVALID_TRANSACTION_BODY precheck.
Now we should run mono code to see if the test passed before we check module code😀

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java]: True unfortunately these tests are not run in mono-service 😄

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Pull request merged**